component/otelcol/processor/resourcedetection/resourcedetection.go

```diff
@@ -0,0 +1,204 @@
+	"github.com/grafana/agent/component/otelcol/processor/resourcedetection/internal/aws/ecs"
+	"github.com/grafana/agent/component/otelcol/processor/resourcedetection/internal/aws/eks"
+	"github.com/grafana/agent/component/otelcol/processor/resourcedetection/internal/aws/elasticbeanstalk"
+	"github.com/grafana/agent/component/otelcol/processor/resourcedetection/internal/aws/lambda"
+	"github.com/grafana/agent/component/otelcol/processor/resourcedetection/internal/azure"
+	"github.com/grafana/agent/component/otelcol/processor/resourcedetection/internal/azure/aks"
+	"github.com/grafana/agent/component/otelcol/processor/resourcedetection/internal/consul"
+	"github.com/grafana/agent/component/otelcol/processor/resourcedetection/internal/docker"
+	"github.com/grafana/agent/component/otelcol/processor/resourcedetection/internal/gcp"
+	"github.com/grafana/agent/component/otelcol/processor/resourcedetection/internal/heroku"
+	kubernetes_node "github.com/grafana/agent/component/otelcol/processor/resourcedetection/internal/k8snode"
+	"github.com/grafana/agent/component/otelcol/processor/resourcedetection/internal/openshift"
+	"github.com/grafana/agent/component/otelcol/processor/resourcedetection/internal/system"
+	otel_service "github.com/grafana/agent/service/otel"
+	"github.com/grafana/river"
+	"github.com/mitchellh/mapstructure"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/processor/resourcedetectionprocessor"
+	otelcomponent "go.opentelemetry.io/collector/component"
+	otelextension "go.opentelemetry.io/collector/extension"
+)
+
+func init() {
+	component.Register(component.Registration{
+		Name:          "otelcol.processor.resourcedetection",
+		Args:          Arguments{},
+		Exports:       otelcol.ConsumerExports{},
+		NeedsServices: []string{otel_service.ServiceName},
+
+		Build: func(opts component.Options, args component.Arguments) (component.Component, error) {
+			fact := resourcedetectionprocessor.NewFactory()
+			return processor.New(opts, fact, args.(Arguments))
+		},
+	})
+}
+
+// Arguments configures the otelcol.processor.probabilistic_sampler component.
+type Arguments struct {
+	// Detectors is an ordered list of named detectors that should be
+	// run to attempt to detect resource information.
+	Detectors []string `river:"detectors,attr"`
+
+	// Override indicates whether any existing resource attributes
+	// should be overridden or preserved. Defaults to true.
+	Override bool `river:"override,attr,optional"`
+
+	// DetectorConfig is a list of settings specific to all detectors
+	DetectorConfig DetectorConfig `river:",squash"`
+
+	// HTTP client settings for the detector
+	// Timeout default is 5s
+	// Client otelcol.HTTPClientArguments `river:",squash"`
+	Timeout time.Duration `river:"timeout,attr,optional"`
+	//TODO: Uncomment this later? Can we just get away with a timeout, or do we need all the http client settings?
+	//      It seems that HTTP client settings are only used in the ec2 detection via ClientFromContext.
```

Comment:
The HTTP settings in the Collector expose [HTTP client settings](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.87.0/processor/resourcedetectionprocessor/config.go#L37-L39) for two reasons:
* The timeout is used to cancel resource detections if they take too long. This use case has nothing to do with HTTP at all.
* The ec2 detection [uses the HTTP settings](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/v0.87.0/processor/resourcedetectionprocessor/internal/aws/ec2/ec2.go#L101-L108) if they are provided. This feature is not documented in the OTel docs.

For now I would like to expose just the `timeout` parameter:
* This is because it would be super confusing to document `timeout` as both a timeout for HTTP and for ec2.
* I'll leave a TODO to implement it in the future.

I wish the OTel Collector worked differently:
* There should be a non-HTTP `timeout` settings for all detectors
* There should be HTTP settings specific to the ec2 detector. That way ec2 won't use `context`, and it will be possible to have different HTTP settings for different detectors.

Maybe it'll be work raising these issues with the upstream maintainers.